### ISSUE TITLE
bundler: failed to load command: rackup (/usr/local/bin/rackup)

### ISSUE THREAD

**Issue by natashadsouza** [StephenOTT/GitHub-Analytics#17]
Hi,

The following command keeps throwing the above mentioned error:
3.3.GITHUB_CLIENT_ID="YOUR CLIENT ID" GITHUB_CLIENT_SECRET="YOUR CLIENT SECRET" bundle exec rackup Get the Client ID and Client Secret from the settings of your created/registered GitHub Application in Step 1.

I tried reinstalling gem and bundle. I am not sure what the issue is. 
Your help is much appreciated!

**Comment by StephenOTT** [comment]
The error is not posted.  Can you repost the error.

I have not used this lib in a long time.  The core functionality is still valid, but possible github's api has changed in places.

**Comment by natashadsouza** [comment]
Hi Stephen, 
I got past the initial error but I still see the following:
<img width="1425" alt="screen shot 2018-08-17 at 3 53 11 pm" src="https://user-images.githubusercontent.com/16967661/44291828-c95bc080-a235-11e8-9a9b-3ae11525859f.png">

**Comment by StephenOTT** [comment]
not sure based on the error.

What are you looking to report on.  Converting this lib to another format so you can report using something like Tableau Community edition would not be much work.  Let me know your use case(s) and i will see what i can do for you

**Comment by natashadsouza** [comment]
We want to analyze our open source project for incoming issues to maximize engineering resources as the team is small. We wanted to use your analytics to create reports for our executive team. 
By the way, most of what you listed in the Readme is exactly what our report is expected to deliver. So great job on listing all those cases!

**Comment by StephenOTT** [comment]
😉 it is probably a bureaucracy you work in then.

do you need the snapshot features? Stuff like: having to create analytics snapshots of the repos, where you are concerned about previous data changing?  or are you just concerned about "whats the state right now?" ?

**Comment by natashadsouza** [comment]
Yes. We are also trying to grow our community, which is why we wanted to use your analytics to generate actionable items. But you are correct! :)

**Comment by StephenOTT** [comment]
@natashadsouza do you have admin access to the repository you want to report on? or could you get access?

**Comment by StephenOTT** [comment]
@natashadsouza okay here is the easy fix that will give you much more data capabilities than this legacy app:

Use the ~new Github Migration API: https://developer.github.com/changes/2018-05-24-user-migration-api/.  (Using the api to download the data does not take very long.  When tested it was just a few minutes max, usually just a few seconds.)

Download your repository and then use Tableau Public: https://public.tableau.com/en-us/s/
With Tableau you can do "all of your reporting needs". 🎉 

When you download the repo, you will get many json files such as:

issues_000001.json
issues_000002.json
issues_000003.json
etc

Use the Tablea Union feature to merge all of these files into a single view:
https://onlinehelp.tableau.com/current/pro/desktop/en-us/union.html

You will then be able to run your reports with Tableau.

You can also Join the Issues with the Issue-Comments json files using the Join feature in Tableau: https://onlinehelp.tableau.com/current/pro/desktop/en-us/joining_tables.html.

----

So the steps would be:

1. Each time you want a snapshot of data you run the migration api. (you can convert this into a small ruby CLI

1. Setup some templated reports on tableau

1. Import the json files from the github migration api.  Use Union to merge the paginated json files. and use the Join feature to make the data connection between the Issues and the Issue-Comments json files. (looks like you will use the `url` field from the Issues json file, and the `issue` field from the issue-comments json file.

**Comment by natashadsouza** [comment]
I do not have it, but I can get admin access if needed.

**Comment by natashadsouza** [comment]
Thank you so much! I will try it out!

**Comment by StephenOTT** [comment]
>I do not have it, but I can get admin access if needed.

You can just create a admin user to do this, and then create a small microservice api to provide the url to download the file.  So if the "company" has issues giving you admin access, you can just create a single isolated user with access.

**Comment by StephenOTT** [comment]
@natashadsouza once you add the two data sources and do the union, you can add the relationship as follows:

![1](https://user-images.githubusercontent.com/1994838/44295006-98e44880-a26e-11e8-956d-558c315b7dc6.png)

![2](https://user-images.githubusercontent.com/1994838/44295009-9b46a280-a26e-11e8-866b-e5bbbfc11fd3.png)

**Comment by StephenOTT** [comment]
@natashadsouza did you make progress?

**Comment by natashadsouza** [comment]
@StephenOTT the migration API keeps throwing me errors. Working on that currently, thanks for checking in!
